packages/docs/docs/concepts/cornerstone-tools/touch.md

```diff
@@ -0,0 +1,156 @@
+    radiusX: number,
+    radiusY: number,
+    force: number,
+    rotationAngle: number,
+  },
+};
+```
+
+## Multitouch Drag Calculations
+
+`TOUCH_DRAG` events have the following structure:
+
+```js
+type TouchDragEventDetail = NormalizedTouchEventDetail & {
+  /** The starting points of the touch event. */
+  startPoints: ITouchPoints,
+  /** The last points of the touch. */
+  lastPoints: ITouchPoints,
+  /** The current touch position. */
+  currentPoints: ITouchPoints,
+  startPointsList: ITouchPoints[],
+  /** The last points of the touch. */
+  lastPointsList: ITouchPoints[],
+  /** The current touch position. */
+  currentPointsList: ITouchPoints[],
+
+  /** The difference between the current and last points. */
+  deltaPoints: IPoints,
+  /** The difference between distances between the current and last points. */
+  deltaDistance: IDistance,
+};
+```
+
+`deltaPoints` is the difference between the mean coordinate point of `lastPointsList` and `currentPointsList`.
+`deltaDistance` is the difference between the average distance between points in `lastPointsList` vs `currentPointsList`
+
+## Usage
+
+You can add an event listener to the element for the event.
+
+```js
+import Events from '@cornerstonejs/tools/enums/Events';
+// element is the cornerstone viewport element
+element.addEventListener(Events.TOUCH_DRAG, (evt) => {
+  // my function on drag
+  console.log(evt);
+});
+
+element.addEventListener('CORNERSTONE_TOOLS_SWIPE', (evt) => {
+  // my function on swipe
+  console.log(evt);
+});
+
+// within the chrome console in a deployed OHIF application
+cornerstone
+  .getEnabledElements()[0]
+  .viewport.element.addEventListener('CORNERSTONE_TOOLS_SWIPE', (evt) => {
+    // my function on swipe
+    console.log('SWIPE', evt);
+  });
+```
+
+A full example can be found by running
+`yarn run example stackManipulationToolsTouch` whose source is [here](https://github.com/gradienthealth/cornerstone3D-beta/blob/gradienthealth/added_touch_events/packages/tools/examples/stackManipulationToolsTouch/index.ts)
+
+## Binding
+
+Touch tools have bindings depending on the number of pointers that are placed down. In the future, bindings can be filter based on force, as well as radius (stylus detection).
+
+```js
+// Add tools to Cornerstone3D
+cornerstoneTools.addTool(PanTool);
+cornerstoneTools.addTool(WindowLevelTool);
+cornerstoneTools.addTool(StackScrollTool);
+cornerstoneTools.addTool(ZoomTool);
+
+// Define a tool group, which defines how mouse events map to tool commands for
+// Any viewport using the group
+const toolGroup = ToolGroupManager.createToolGroup(toolGroupId);
+
+// Add tools to the tool group
+toolGroup.addTool(WindowLevelTool.toolName);
+toolGroup.addTool(PanTool.toolName);
+toolGroup.addTool(ZoomTool.toolName);
+toolGroup.addTool(StackScrollTool.toolName);
+
+// Set the initial state of the tools, here all tools are active and bound to
+// Different mouse inputs
+toolGroup.setToolActive(ZoomTool.toolName, {
```

Comment:
and how many touchpoints we support? (given that the device supports it), can someone register a 5 touch point bingind?

Reply:
We can support as many touchpoints as the hardware device so 5 touch points are possible. Fast follows: we can filter on radius / force for touch pen devices.

Reply:
can we please modify and put that info in the .md doc then?

Reply:
yeah we can add that to the conceptual doc.